CHANGELOG.md

```diff
@@ -1,3 +1,41 @@
+# to-be-released
+
+## Changed
+
+- [BREAKING] Internal representation of hash schemas was changed to be a simple list of key types (flash-gordon)
+  `Dry::Types::Hash#with_type_transform` now yields a key type instead of type + name:
+  ```ruby
+  Dry::Types['stirct.hash'].with_type_transform { |key| key.name == :age ? key.required(false) : key }
+  ```
+
+## Added
+
+- Optional keys for schemas can be provided with ?-ending symbols (flash-gordon)
+  ```ruby
+  Dry::Types['stirct.hash'].schema(name: 'strict.string', age?: 'strict.integer')
+  ```
+- Another way of making keys optional is setting `required: false` to meta. In fact, it is the preferrable
+  way if you have to store this information in `meta`, otherwise use the Key's API (see below) (flash-gordon)
```

Comment:
@flash-gordon In reading this, what's not clear to me is _why_ this is preferable in this case. Is it because you're already adding to the meta so it's easier just to add that extra `required: false` entry at the same time? Is this _really_ preferable here, or is it more it's just somewhat more _convenient?_

Reply:
because consistency, we now use `required` under the hood and don't rely on fragile metadata so naming things straight is the preferable way. `:required` from `meta` goes to `attr_reader :required` of `Key`